components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/ForwardingGroupPolicyProvider.kt

```diff
@@ -70,8 +79,42 @@ internal class ForwardingGroupPolicyProvider(coordinatorFactory: LifecycleCoordi
         }
 
         val trustedCertificates = groupPolicy.p2pParameters.tlsTrustRoots.toList()
+        val sessionPkiMode = groupPolicy.p2pParameters.sessionPki
+
+        val sessionTrustStore = groupPolicy.p2pParameters.sessionTrustRoots?.let { convertToKeyStore(it) ?: return null }
+        if (sessionTrustStore == null && sessionPkiMode != P2PParameters.SessionPkiMode.NO_PKI) {
+            logger.warn("Session trust roots is unexpectedly null in the group policy for $holdingIdentity.")
+        }
 
-        return GroupPolicyListener.GroupInfo(holdingIdentity, networkType, protocolModes, trustedCertificates)
+        return GroupPolicyListener.GroupInfo(
+            holdingIdentity,
+            networkType,
+            protocolModes,
+            trustedCertificates,
+            sessionPkiMode,
+            sessionTrustStore
+        )
     }
 
+    private fun convertToKeyStore(pemCertificates: Collection<String>): KeyStore? {
```

Comment:
we could use the `typealias` here
```suggestion
    private fun convertToKeyStore(pemCertificates: Collection<PemCertificateChain>): KeyStore? {
```